bowtie/_cli.py

```diff
@@ -146,53 +146,55 @@ def implementation_subcommand(reporter: _report.Reporter = SILENT):
     """
 
     def wrapper(fn: ImplementationSubcommand):
-        async def run(
```

Comment:
I don't understand immediately why you refactored this function. It seems like it should do exactly what you need, and your job is to just provide it with the right `image_names`, which is how I intended it to be used. Will have to try to understand why you changed it.

Reply:
Yes @Julian you are indeed correct but if you look closely in my code I am still providing the `run` function with the right image names as you pointed out. The only refactoring I did was to take that function(`run`) out of its `wrapper` function and rename it as `_run_implementations` and make it a separate one so that I can call it again inside the `all_implementations_subcommand` function and don't have to rewrite it in there. 

Now why I took it out separately is because in the `all_implementations_subcommand` function (which I introduced in my code), the `wrapper` function inside it isn't really expecting the `@IMPLEMENTATION` option /decorator on it unlike the `implementation_subcommand` function's `wrapper` function which does have that decorator on it.

 The user when using the `bowtie filter-implementations` command only provides either the `dialect` or the `language` as you had mentioned in the issue description and in this case even if we use `@IMPLEMENTATION` on the wrapper function it doesn't really make sense because we aren't expecting any implementations option from the user side and it will throw error because it is set as required = True. When the user hits this command we simply start off all the containers (which is why its slow) and filter out based on either the provided dialect or the language or both.
I also thought of doing `@IMPLEMENTATION(required=False)` but then when the user does `bowtie filter-implementations --help` he / she will still see that the command expects image names.